pkg/client/query/meteredquerier.go

```diff
@@ -0,0 +1,57 @@
+package query
+
+import (
+	"context"
+	"time"
+
+	"github.com/cosmos/gogoproto/grpc"
+	"github.com/pokt-network/poktroll/pkg/relayer"
+	googlegrpc "google.golang.org/grpc"
+)
+
+const (
+	// ComponentCtxKey is the context key used to identify the component calling the gRPC method
+	// using the context.WithValue method.
+	ComponentCtxKey = "component"
+	// ComponentCtxProxy is the identifier for the proxy component in the relayer.
+	ComponentCtxProxy = iota
+	// ComponentCtxMiner is the identifier for the miner component in the relayer.
+	ComponentCtxMiner
+	// ComponentCtxSessionsManager is the identifier for the sessions manager component in the relayer.
+	ComponentCtxSessionsManager
+)
+
+// componentKindNames maps component context keys to their string representations.
+// This is used to provide human-readable names for the components in metrics and logging.
+var componentKindNames = map[int]string{
+	ComponentCtxProxy:           "proxy",
+	ComponentCtxMiner:           "miner",
+	ComponentCtxSessionsManager: "sessions_manager",
+}
+
+// meteredClientConn is a wrapper around grpc.ClientConn that captures the duration of gRPC calls.
+// It implements the grpc.ClientConn interface and is used to monitor the performance of gRPC calls
+// by recording the time taken for each call.
+type meteredClientConn struct {
```

Comment:
This is VERY similar to the work @adshmh is doing here: https://github.com/buildwithgrove/path/pull/383/files

See `httpClientWithDebugMetrics`

Wdyt of calling it `grpcConnWithDebugMetrics`?

It'll set things up to add more debug metrics along thew ay.